lib/main.js

```diff
@@ -0,0 +1,244 @@
+"use strict"
+const core = require('@actions/core');
+const exec = require('child_process').exec;
+const github = require('@actions/github');
+const assert = require('assert');
+const fetch = (...args) => import('node-fetch').then(({default: fetch}) => fetch(...args));
+
+// TEST CODE
+function getTime() {
+  // test datetime code for timing testing
+  var today = new Date();
+  var time = today.getHours() + "" + today.getMinutes() + today.getSeconds();
+  return time
+}
+
+function setInputs() {
+  const setInput = (name, value) =>
+    process.env[`INPUT_${name.replace(/ /g, '_').toUpperCase()}`] = value;
+
+  // setInput("appName", "amber-test-container")
+  setInput("appName", "error-test")
+  setInput("resourceGroup", "amber-test-container")
+  // setInput("imageID", "mcr.microsoft.com/azuredocs/containerapps-helloworld:latest")
+  setInput("imageID", "ambertest.azurecr.io/testapp:errors")
+  setInput("logAnalyticsWorkspace", "amber-test-app-logs")
+  setInput("revisionSuffix", getTime())
+  setInput("stepPct", "50")
+  setInput("stepTime", "5")
+  setInput("monitorInterval", "1")
+  setInput("canaryDeploy", "true")
+  setInput("finalPct", "100")
+  setInput("monitors", "https://www.google.com")
+  setInput("errorThreshold", "0")
+}
+setInputs() // for local runs only to set environment variables
+// END TEST CODE
+
+// grab input values from github environment
+const APP = core.getInput('appName') // REQD application name
+const RG = core.getInput('resourceGroup') // REQD resource group name
+const IMAGE = core.getInput('imageID') // REQD image to update with
+const REV_SUFFIX = core.getInput('revisionSuffix') // REQD new revision suffix
+const LOG_ANALYTICS = core.getInput('logAnalyticsWorkspace') // log analytic workspace app is in
+const CANARY = core.getBooleanInput('canaryDeploy') // whether to use canary deployment
+const STEP_PCT = parseInt(core.getInput('stepPct')) // step pct of traffic going to new revision
+const STEP_TIME = parseFloat(core.getInput('stepTime')) // step time between traffic shifts
+const FINAL_PCT = parseInt(core.getInput('finalPct')) // final pct of traffic going to new revision
+const MONITORS = core.getInput('monitors').replace(/ +/g, '').split(/,|;|\n/) // api endpoints to hit
+const MONITOR_INTERVAL = parseFloat(core.getInput('monitorInterval')) // time between monitoring
+const ERROR_THRESHOLD = parseFloat(core.getInput('errorThreshold')) // max error tolerance on http pings
+const TIME_CONVERSION = 60000 // conversion factor from minutes to milliseconds
+
+let RESOURCE = ""
+let prevTrafficSettings = ""
+
+// utility function for async exec command
+const execute = function (command) {
+  return new Promise((resolve, reject) => {
+    console.log(`Running command ${command} at ${getDateTime()}`)
```

Comment:
Good to create a separate logging class and method something like 

class Logger {
 static LogInfo(message) { ... }
 static LogError(exception) { ... }
  static #privateLogInternal(params) { ... }
}

These methods can call an common Private method to log the events in a more uniform structure like:
2022-28-07 11:47Z      INFO     Running command foobar ...
2022-28-07 11:48Z      INFO     Some more log message here.
2022-28-07 11:48Z      ERROR  Exception encountered - some exception here.


Now, you can use this logger in your main code like this:

Logger.LogInfo(`running command ${cmd}`);
Logger.LogError(`{error}`);